bioptim/dynamics/dynamics_functions.py

```diff
@@ -676,7 +676,9 @@ def apply_parameters(parameters: MX.sym, nlp):
             The definition of the system
         """
 
+        offset = 0
         for param in nlp.parameters:
             # Call the pre dynamics function
             if param.function:
-                param.function(nlp.model, parameters, **param.params)
+                param.function(nlp.model, parameters[offset : offset + param.size], **param.params)
+                offset += param.size
```

Comment:
I assume this ensure that the user only gets the MX parameters associated with this particular parameter... It feels to right thing to do, but at the same time I feel that the user may wants to have access to other values? It would mean that this information should be somehow passed to the user.. Maybe if they need it, they may just declare larger parameter..

Reply:
> Maybe if they need it, they may just declare larger parameter..
With the potential problem of free variables.
I propose we stick to this proposal for now (to my knowledge, we never encountered such a situation yet).